kernel/kernel-api/src/main/java/io/delta/kernel/Scan.java

```diff
@@ -217,6 +231,30 @@ public FilteredColumnarBatch next() {
                             "Column mapping mode is not yet supported: " + columnMappingMode);
                 }
 
+                // Type widening: add casts whenever the type of column read in the parquet file
+                // doesn't match the expected type from the table schema.
+                for (int colIdx = 0; colIdx < logicalReadSchema.length(); colIdx++) {
+                    StructField field = logicalReadSchema.at(colIdx);
+                    DataType parquetType =
```

Comment:
Please add a comment saying the look is done after the column mapping is done (if any) on the physical data batch.